src/util/FitsComparator.cpp

```diff
@@ -0,0 +1,208 @@
+//
+// Created by Samuel Cappon on 3/5/20.
+//
+
+#include <util/FitsComparator.h>
+#include <fitsio.h>
+#include <core/Exceptions.h>
+#include <iostream>
+#include <memory>
+#include <vector>
+#include <map>
+#include <algorithm>
+#include <iterator>
+#include <core/SimpleLogger.h>
+
+using namespace std;
+
+FitsComparator::FitsComparator(string fitsfile1, string fitsfile2) {
+    int status = 0;
+    int hdunum;
+
+        fits_open_file(&_fptr, fitsfile1.c_str(), READONLY, &status);
```

Comment:
This concerns me. If your object throws error while executing something, _fptr and _fptr2 will be dangling pointers. Should use RAII or std::unique_ptr for this.